src/arizona_template_compiler.erl

```diff
@@ -0,0 +1,653 @@
+    norm_assigns_vars_1(maps:to_list(NormAssigns), ChangeableVars).
+
+norm_assigns_vars_1([{ChangeableVar, #{vars := Vars}} | T], ChangeableVars) ->
+    Indexes = proplists:get_all_values(ChangeableVar, ChangeableVars),
+    norm_assigns_vars_2(Vars, ChangeableVar, Indexes, T, ChangeableVars);
+norm_assigns_vars_1([], _) ->
+    [].
+
+norm_assigns_vars_2([Var | Vars], ChangeableVar, Indexes, T, ChangeableVars) ->
+    norm_assigns_vars_3(Indexes, Var, Vars, ChangeableVar, Indexes, T, ChangeableVars);
+norm_assigns_vars_2([], _, _, T, ChangeableVars) ->
+    norm_assigns_vars_1(T, ChangeableVars).
+
+norm_assigns_vars_3([Index | Indexes], Var, Vars, ChangeableVar, AllIndexes, T, ChangeableVars) ->
+    [{Var, Index}
+     | norm_assigns_vars_3(Indexes, Var, Vars, ChangeableVar, AllIndexes, T, ChangeableVars)];
+norm_assigns_vars_3([], _, Vars, ChangeableVar, Indexes, T, ChangeableVars) ->
+    norm_assigns_vars_2(Vars, ChangeableVar, Indexes, T, ChangeableVars).
+
+stateful_state(Mod, Fun, Macros, State) ->
+    State#state{
+        module = Mod,
+        function = Fun,
+        macros = Macros,
+        index = 0,
+        path = [State#state.index | State#state.path]
+    }.
+
+stateless_state(Macros, Attrs, State) ->
+    State#state{
+        macros = maps:merge(State#state.macros, Macros),
+        attributes = Attrs
+    }.
+
+incr_index(N, #state{index = Index} = State) ->
+    State#state{index = Index + N}.
+
+%% --------------------------------------------------------------------
+%% EUnit
+%% --------------------------------------------------------------------
+
+-ifdef(TEST).
+-compile([export_all, nowarn_export_all]).
+-include_lib("eunit/include/eunit.hrl").
+
+compile_test() ->
+    ?assertMatch({ok,
+         #{function := render, id := [0], module := arizona_template_compiler,
+           changeable :=
+            #{0 :=
+               {block,
+                #{function := render, id := [0], module := arizona_template_compiler,
+                  changeable :=
+                   #{0 :=
+                      {expr, #{function := _, id := [0, 0], vars := [count]}}},
+                  static :=
+                   [<<"<div><span>Count:">>,
+                    <<"</span><button type=\"button\" "
+                      "onclick=\"arizona.send.bind(this)('incr')\">",
+                     "Increment</button></div>">>],
+                  changeable_vars := #{count := [[0, 0]]},
+                  norm_assigns_vars := [{count, [0, 0]}],
+                  changeable_indexes := [0],
+                  norm_assigns :=
+                   #{count := #{function := _, vars := [count]}}}},
+              1 :=
+               {block,
+                #{function := render, id := [1], module := arizona_template_compiler,
+                  changeable :=
+                   #{0 :=
+                      {expr, #{function := _, id := [1, 0], vars := [count]}}},
+                  static :=
+                   [<<"<div><span>Count:">>,
```

Comment:
The id is missing in the block element:
```suggestion
                   [<<"<div arizona-id=\"[1]\"><span>Count:">>,
```

Reply:
[0a36dbc](https://github.com/arizona-framework/arizona/pull/164/commits/0a36dbc153e88eb72935db9e6de4d50b53530f45)